build/authors.py

```diff
@@ -2,9 +2,119 @@
 Process author table
 """
 
-import collections
+from collections import OrderedDict
+
 import pandas
+import numpy as np
+
+
+def initials(full_name):
+    """
+    Return author intials in the form F.L. for First Last.  Accomodates an
+    aribtrary number of whitespace-separated strings.  Converts periods to
+    whitespace so that multiple middle initials are retained.
+    """
+    full_name = full_name.replace('.', ' ')
+    name_tokens = full_name.strip().split()
+    # Keeps only the first part of hyphenated names
+    name_tokens = [name[0] + '.' for name in name_tokens]
+    return ''.join(name_tokens)
+
+def format_list(tokens):
+    """
+    Format a list of string tokens such that a pair [X, Y] is formatted 'X and
+    Y' and a list [X, Y, Z] becomes 'X, Y, and Z'.  Single tokens are returned
+    as-is.
+    """
+    if len(tokens) <= 2:
+        return ' and '.join(tokens)
+    else:
+        tokens[-1] = 'and ' + tokens[-1]
+        return ', '.join(tokens)
+
+def get_funding(author_df):
+    """
+    Returns a dictionary mapping funders ('GBMF', 'NIH', 'NSF', 'OTHER') to
+    award and author text.  An author may have more than one funding
+    source and multiple authors may have the same funding source.
+    """
+    # Initially create a map from funders to dictionaries that map an award to
+    # the authors supported by that award
+    # funding_info data structure has the form
+    # {funder1: {award1: [author1], award2: [author2, author3]}}
+    funding_info = OrderedDict()
+    for (_, row) in author_df.dropna(subset=['funding']).iterrows():
+        for grant in row['funding'].split(','):
+            # Separate into the funder and the award
+            grant_tokens = grant.strip().split(None, 1)
+            funder_map = funding_info.setdefault(grant_tokens[0], dict())
+            funder_map.setdefault(grant_tokens[1], list()).append(row['initials'])
+
+    formatted_info = OrderedDict()
+    for funder in funding_info.keys():
+        # The map from awards to author lists for a single funder
+        funder_map = funding_info[funder]
+        funder_awards = list()
+        for award in sorted(funder_map.keys()):
+            authors = sorted(funder_map[award])
+            funder_awards.append(f'{award} ({format_list(authors)})')
+        formatted_info[funder] = format_list(funder_awards)
 
+    return formatted_info
+
+def get_affiliations(author_df):
```

Comment:
Yes, I see your point.  That is more readable.

I am short on time, so I suggest we merge now and make a mental note that the code could be improved by rewriting in pandas.

I'll restore `.gitignore` and remove `index.html` before merging.